### PR TITLE
Test/crud tests

### PR DESCRIPTION
**Pull request by gorold**
### Context
More test cases are required to ensure that API works well

### Changes
- Added some new test cases for the CRUD API
- Fixed some bugs related to changing User's pk from ```email``` to ```id```

### Considerations
- Edited the InviteUser view to use UserCreateSerializer to validate the data, using the code pattern from DRF official sources, can check if this is better. (Test cases were previously causing crashes)

### CODE REVIEW

File: secretsauce/apps/account/views.py [hunk 1]
```diff
@@ -33,26 +33,15 @@ class InviteUser(generics.CreateAPIView):
...
+        if serializer.is_valid():
+            serializer.save()
+            mappings = {
+            'email': serializer.data.get('email'),
+            'password':  serializer.data.get('password'),
+            }
+            send_email('You have been invited to RMS Pricing Analytics Platform!', 'donotreply@rmsportal.com', [mappings['email']], '', 'create_user.html', mappings)
+
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```
Comment (eugenechia95) [inline secretsauce/apps/account/views.py:47]: Did you test the endpoint directly? Can't get it to work.

Seems like the request.data dictionary cant be modified? Can clone `request.data` and pass it into the serializer.
Also, frontend will be passing the company_id over. I am not sure whether the drf serializer will be able to match the company_id to the company instance
image

Comment (eugenechia95) [reply secretsauce/apps/account/views.py:47]: Also found the UserCreateSerializer source code if u are interested
https://github.com/sunscrapers/djoser/blob/6ed0a05522234e56b3c9b9ad06fc08079a73984c/djoser/serializers.py

Comment (gorold) [reply secretsauce/apps/account/views.py]: Fixed the error, should be able to work now.
Serializer matches the company.id since its the primary key.

Comment (eugenechia95) [reply secretsauce/apps/account/views.py]: `serializer.data.get('password')` does not return the password as i think the serializer does not return the password. Fixed the bug alr. Good to go.

**Pull request merged**